src/giskardpy/python_interface/python_interface.py

```diff
@@ -1856,3 +2116,54 @@ def get_result(self, timeout: rospy.Duration = rospy.Duration()) -> MoveResult:
 
     def _feedback_cb(self, msg: MoveFeedback):
         self.last_feedback = msg
+
+    def get_end_motion_reason(self, move_result: MoveResult = None, show_all=False):
+        if not move_result and not self.last_execution_state:
+            raise Exception('No MoveResult available to analyze')
+        elif not move_result:
+            execution_state = self.last_execution_state
+        else:
+            execution_state = move_result.execution_state
+
+        result = {}
+
+        endMotion_idx = 0
+        cancelMotion_idx = 0
+        idx = 0
+        for monitor in execution_state.monitors:
+            if monitor.monitor_class == 'CancelMotion':
+                cancelMotion_idx = idx
+            if monitor.monitor_class == 'EndMotion':
+                endMotion_idx = idx
+            idx += 1
+
+        if execution_state.monitor_state[endMotion_idx] == 1:
+            # the end motion was successful
+            return result
+
+        if show_all:
+            for monitor, state in zip(execution_state.monitors, execution_state.monitor_state):
+                result[monitor.name] = state
```

Comment:
`{monitor.name: state for monitor, state in zip(execution_state.monitors, execution_state.monitor_state)}`